hack/update-swagger-spec.sh

```diff
@@ -37,7 +37,10 @@ make -C "${KUBE_ROOT}" WHAT=cmd/kube-apiserver
 
 function cleanup()
 {
-    [[ -n ${APISERVER_PID-} ]] && kill ${APISERVER_PID} 1>&2 2>/dev/null
```

Comment:
I suspect that `kill` returned 1 and the script was failing (because we also use `errexit` option). Anyway, using `&&` with `errexit` also a bad practice as it also could return 1.

Reply:
The other thing you might do here is replace line 50 below:

from:
```
trap cleanup EXIT SIGINT
```

to:
```
kube::util::trap_add cleanup EXIT
```

This would make the trap resilient to breaking if multiple traps are added (like creating a temp dir)

Reply:
In your example, why `SIGINT` is missing? If we need it, looks like it will require to execute method twice:
```bash
kube::util::trap_add cleanup EXIT
kube::util::trap_add cleanup SIGINT
```
What I'm missing here?

Also, I'd like to merge it as-is because I don't see a benefit from using `kube::util::trap_add` here. But if you will insist on that, I wouldn't mind of course :)

Reply:
Trapping both signals is redundant. EXIT will trigger on a error, or on a SIGINT. EXIT is all that is needed.

The reason I'd suggest adding this now, is the original problem (things not being cleaned up) will be better prevented from breaking in the future with this. If we (for example) add tmpdir code to the goenv or etcd libraries, then they may overwrite the trap.

Reply:
@cblecker I've just pushed a commit that adds `kube::util::trap_add`. PTAL.